### PR TITLE
Updated Jupyterlab

### PR DESCRIPTION
**Pull request by GTrunSec**
fixes: #440

### CODE REVIEW

**Comment by garbas** [comment]
@GTrunSec Can you only include https://github.com/tweag/jupyenv/pull/451/commits/f13ff5d7f05c20a8223fec510d451f0d578e4bc7 commit and not the rest?

**Comment by GTrunSec** [comment]
@garbas  sure.

**Review by garbas — APPROVED** [review]
Looks good to me, except the preferWheels, change.

I've looked at [the release notes since jupyterlab 3.5.0][1] and found this two PRs might be cause some problems since they touch some cirectories stuctures:
- https://github.com/jupyterlab/jupyterlab/pull/13589
- https://github.com/jupyterlab/jupyterlab/pull/13788

At the same time I found an interesting PR we should be keeping an eye on since it will allow us to have less hacky per environment config: https://github.com/jupyterlab/jupyterlab/issues/12916

[1]: https://jupyterlab.readthedocs.io/en/stable/getting_started/changelog.html

File: lib/jupyter.nix [hunk 1]
```diff
@@ -23,7 +23,7 @@
     python ? pkgs.python3,
     editablePackageSources ? {},
     extraPackages ? (ps: []),
-    preferWheels ? false,
+    preferWheels ? true,
```
[the remark this answers is not in the document]
Comment (GTrunSec) [reply lib/jupyter.nix]: > cause some weird problems

So hard to resolve some dependencies. We can override some packages for building without preferWheels, such as `jupyterlab, jupyter -server`

**Pull request merged**